test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -0,0 +1,2018 @@
+
+        /******************************************************************************************/
+        /*** Remove Liquidity                                                                   ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -7);
+
+        DecreasePositionResult memory decreaseResult = _decreasePosition(increaseResult.tokenId, 1_000_000_000e6, 0, 0);
+
+        assertEq(decreaseResult.amount0Received, 0);
+        assertEq(decreaseResult.amount1Received, 9_449_821.223797e6);
+        assertEq(decreaseResult.amount0Received + decreaseResult.amount1Received, 9_449_821.223797e6); // Lost 0 USD.
+    }
+
+    function test_uniswapV4_attack_priceAboveToBelow_wideTicks() public {
+        // Setup the pool and the controller.
+        vm.startPrank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -200, 200, 200); // Allow wider tick range.
+        rateLimits.setRateLimitData(_DEPOSIT_LIMIT_KEY,  20_000_000e18, uint256(20_000_000e18) / 1 days);
+        rateLimits.setRateLimitData(_WITHDRAW_LIMIT_KEY, 20_000_000e18, uint256(20_000_000e18) / 1 days);
+        vm.stopPrank();
+
+        /******************************************************************************************/
+        /*** Frontrun                                                                           ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -7);
+
+        uint256 amountOut1 = _externalSwap(_alice, address(usdc), 3_020_000e6);
+
+        /******************************************************************************************/
+        /*** Add Liquidity (Current price was expected to be above the range, but is below)     ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -501);
+
+        IncreasePositionResult memory increaseResult = _mintPosition(-200, -10, 1_000_000_000e6, type(uint160).max, type(uint160).max);
+
+        assertEq(increaseResult.amount0Spent, 9_549_562.082877e6); // Expected 0
+        assertEq(increaseResult.amount1Spent, 0); // Expected 9_449_821.223798e6
+        assertEq(increaseResult.amount1Spent + increaseResult.amount0Spent, 9_549_562.082877e6); // Expected 9_449_821.223798e6
+
+        /******************************************************************************************/
+        /*** Backrun                                                                            ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -501);
+
+        uint256 amountOut2 = _externalSwap(_alice, address(usdt), uint128(amountOut1));
+
+        assertEq(amountOut2, 3_067_685.526025e6);
```

Comment:
These should be moved into a dedicated section for user actions where assertions are also performed

Reply:
Other functions should be consolidated in here as well